Exoplanet tweaks, reenabled on Torch

comma opened the pull request:
Includes #19889 (#19743)
Makes exoplanet size not hardcoded.
Adds zen check to wild plants/vines to stop processing when no one cares.

mustafakalash committed:
Makes exoplanets smaller and reenables them on the torch.

mustafakalash committed:
Makes flammable exoplanets rare.

comma committed:
Tweaks to exoplanets

Shortens fireproofing code

Makes size of exoplanets setuppable per map rather than hardcoded limit.

Makes plants not tick on empty zlevels

Invisible soil trays (aka vines or plants in the wild) do not process
now if there's no client-bearing living mob on their zlevels.
Plants on station zlevels always tick withotu checking that.

Adds observer check to simple animal life too
Simple animals now won't Llife() if no one's there to look at them

Makes border generation actually work
Only add borderwalls when needed (non-fullsized planets)

Also reshuffles conditions in Entered or exoplanet turfs so we're
grabbing overmap things on non overmap maps.

Makes all exoplanet turfs use carpet footsteps for now. At least better
than fucking plating ones.

Makes vines actually fall asleep. They always wanted to place down a plant tray so they never did, now checks properly if they CAN do that.
Cuts down on some unnecessary icon updates for vines.

Bumps size up, perfomance isn't that different on this scale. 

Makes overmap smaller to find fun things easier, adjusts number of event accordingly.

comma commented:
Need to properly do some test runs once I'm not on potato, in ~7 hours, opening for code stuff spotting I guess

Techhead0 commented on code/modules/hydroponics/trays/tray_soil.dm:
```diff
@@ -39,6 +39,16 @@
+		return ..()
+	var/list/zlevels = GetConnectedZlevels(z)
+	for(var/client/C) //if a tree ticks on the empty zlevel does it really tick
+		if(isliving(C.mob)) //(no it doesn't)
```
Isn't there a list of cliented living mobs?

EDIT: Yeah, it's `GLOB.player_list`

comma replied:
Naw, there's list of all mobs with clients, or list of all living mobs, but not both. So looping over first would be same thing more or less, second even more mobs to cover.

atlantiscze commented on code/modules/hydroponics/trays/tray_soil.dm:
```diff
@@ -39,6 +39,16 @@
+/obj/machinery/portable_atmospherics/hydroponics/soil/invisible/Process()
+	if(z in GLOB.using_map.station_levels) //plants on station always tick
+		return ..()
+	var/list/zlevels = GetConnectedZlevels(z)
```
As this is in process, it will probably be called many times for many objects. Consider moving it to Initialize and a variable instead of calculating it every tick.

I assume plants don't ever move, but if yes, the cached information could be invalidated when move is called.

comma committed:
Some soundstuff

Adds footstep sounds for grass and water.
Adds chop sound to hatchets/machetes.
Vines/plants now make sound (weapon's) when hit.

comma commented:
Okay that will do

comma committed:
Lets exoplanet use maploading ruins

Converts monolith ring to a maploading ruing rather than randomgenned thing.

comma commented:
Not sure what's up with those rogue line endings on the away site generation, botched conflict resolution probably.

mattmatterson111 commented on code/modules/hydroponics/trays/tray_soil.dm:
```diff
@@ -25,6 +25,7 @@
 	name = "plant"
 	icon = 'icons/obj/seeds.dmi'
 	icon_state = "blank"
+	var/list/connected_zlevels //cached for checking if we someone is obseving us so we should process 
```
"if we someone is observing"

Super minor but it still stuck out to me.

Anton-Kr replied:
pls this is litterally a comment

comma replied:
yeah gonna go ahead and say that not going to bother.

MistakeNot4892 replied:
ZERO RESOLVE.

PurpleMartinJCK replied:
Dunno I've been pinged by @Techhead0 for typos in comments so there's precedent.

Techhead0 replied:
I only make a fuss about counterfactual comments. Typos are fine.

PsiOmegaDelta commented on code/_helpers/game.dm:
```diff
@@ -26,6 +26,15 @@
 
+/proc/living_observers_present(var/list/zlevels)
+	if(LAZYLEN(zlevels))
+		for(var/client/C) //if a tree ticks on the empty zlevel does it really tick
```
Because BYOND is weird about when a client exists or not you should do `in clients `

PsiOmegaDelta commented on code/_helpers/game.dm:
```diff
@@ -26,6 +26,15 @@
+		for(var/client/C) //if a tree ticks on the empty zlevel does it really tick
+			if(isliving(C.mob)) //(no it doesn't)
+				var/mob/M = C.mob
+				if(M.z in zlevels)
```
This will miss all clients in mobs inside another object.
You need get_turf()

comma commented:
okay third time was not the charm

comma committed:
Small fixies

Robusitfied zen check
Checks that turf indeed exists before turning it into a border
Fixes rogue line endings

PsiOmegaDelta commented on code/modules/hydroponics/spreading/spreading_growth.dm:
```diff
@@ -122,9 +123,24 @@
+	if(targets_in_range()) //got someone to grab
+		return FALSE
+	if(parent == src && !plant) //should settle down and spawn a tray
+		var/turf/simulated/T = get_turf(src)
```
Any reason this is a simulated turf?

comma replied:
Dunno, I just moved older code here. @FTangSteve ???

Techhead0 replied:
Plants interact with gasses to varying degrees. That might be why?

PurpleMartinJCK replied:
Yeah they use gasses and stuff to know for health and survival.

PsiOmegaDelta replied:
The point of concern is the CanZPass-proc.
If it doesn't exist on atom or turf level there is a potential runtime here.

Techhead0 replied:
`CanZPass` exists at the turf level:
https://github.com/Baystation12/Baystation12/blob/b5aa9808dced155f7f223a3afca8e3e269196f3e/code/modules/multiz/turf.dm#L1

Techhead0 replied:
`CanZPass` exists at the turf level:
https://github.com/Baystation12/Baystation12/blob/b5aa9808dced155f7f223a3afca8e3e269196f3e/code/modules/multiz/turf.dm#L1
